src/main/java/com/faforever/client/teammatchmaking/MatchmakingQueueItemController.java

```diff
@@ -0,0 +1,151 @@
+import com.faforever.client.i18n.I18n;
+import com.faforever.client.main.event.ShowMapPoolEvent;
+import com.faforever.client.player.PlayerService;
+import com.google.common.eventbus.EventBus;
+import javafx.animation.KeyFrame;
+import javafx.animation.Timeline;
+import javafx.beans.value.ChangeListener;
+import javafx.beans.value.ObservableValue;
+import javafx.event.ActionEvent;
+import javafx.scene.Node;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ToggleButton;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.time.Duration;
+import java.time.Instant;
+
+import static javafx.beans.binding.Bindings.createBooleanBinding;
+import static javafx.beans.binding.Bindings.createStringBinding;
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class MatchmakingQueueItemController implements Controller<Node> {
+
+  private final static String QUEUE_I18N_PATTERN = "teammatchmaking.queue.%s.%s";
+
+  private final PlayerService playerService;
+  private final TeamMatchmakingService teamMatchmakingService;
+  private final I18n i18n;
+  private final EventBus eventBus;
+
+  public Node queueItemRoot;
+  public Label playersInQueueLabel;
+  public Label queuePopTimeLabel;
+  public ToggleButton joinLeaveQueueButton;
+  public Label refreshingLabel;
+  public Label matchFoundLabel;
+  public Label matchStartingLabel;
+  public Label matchCancelledLabel;
+  public Button mapPoolButton;
+
+
+  private Timeline queuePopTimeUpdater;
+
+  private MatchmakingQueue queue;
+
+  public MatchmakingQueueItemController(PlayerService playerService, TeamMatchmakingService teamMatchmakingService, I18n i18n, EventBus eventBus) {
+    this.playerService = playerService;
+    this.teamMatchmakingService = teamMatchmakingService;
+    this.i18n = i18n;
+    this.eventBus = eventBus;
+  }
+
+  @Override
+  public void initialize() {
+    eventBus.register(this);
+    joinLeaveQueueButton.widthProperty().addListener(new ChangeListener<Number>() {
+      @Override
+      public void changed(ObservableValue<? extends Number> observable, Number oldValue, Number newValue) {
+        if ((double) newValue > 150.0) {
+          joinLeaveQueueButton.setText(
+              i18n.get(String.format(QUEUE_I18N_PATTERN, queue.getQueueName(), "fullName")));
+        } else {
+          joinLeaveQueueButton.setText(
+              i18n.get(String.format(QUEUE_I18N_PATTERN, queue.getQueueName(), "shortName")));
+        }
+      }
+    });
+  }
+
+  @Override
+  public Node getRoot() {
+    return queueItemRoot;
+  }
+
+  void setQueue(MatchmakingQueue queue) {
+    this.queue = queue;
+
+    playersInQueueLabel.textProperty().bind(createStringBinding(
+        () -> i18n.get("teammatchmaking.playersInQueue", queue.getPlayersInQueue()).toUpperCase(),
```

Comment:
Isn't the toUpperCase something that should be configurable in the control via css or something?